projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/structlog_plugin.py

```diff
@@ -77,6 +77,66 @@ def process(self, msg, kwargs):
 """
 
 
+def _get_syslog_config(context: CoreContext):
+    syslog_enabled = context.configuration.get_value(SYSLOG_ENABLED_KEY)
+    syslog_host = context.configuration.get_value(SYSLOG_HOST_KEY)
+    syslog_port = context.configuration.get_value(SYSLOG_PORT_KEY)
+    syslog_protocol = context.configuration.get_value(SYSLOG_PROTOCOL_KEY)
+
+    return syslog_enabled, syslog_host, syslog_port, syslog_protocol
+
+
+def _create_formatter_and_metadata_filter(context: CoreContext):
+    metadata_keys = context.configuration.get_value(STRUCTLOG_LOGGING_METADATA_KEY)
+    logging_formatter = context.configuration.get_value(STRUCTLOG_LOGGING_FORMAT_KEY)
+    custom_format_string = context.configuration.get_value(
+        STRUCTLOG_CONSOLE_LOG_PATTERN
+    )
+
+    formatter, metadata_filter = create_formatter(
+        logging_formatter, metadata_keys, custom_format_string
+    )
+    return formatter, metadata_filter
+
+
+def _configure_non_root_log_levels(context: CoreContext):
+    if logging.getLogger().getEffectiveLevel() == logging.NOTSET:
+        log_level = os.environ.get(
+            "LOG_LEVEL_VDK", os.environ.get("VDK_LOG_LEVEL_VDK", "INFO")
```

Comment:
Why are we using environment varaible and not `context.configuration`?

Reply:
Not sure, I copied this over. I'll change it and test before we merge.

Reply:
I think it's because we're overriding a core option. Not really sure. @yonitoo can you please help?

Reply:
While I was transferring the functionality from vdk-core back then, I think there was a case where the context config LOG_LEVEL_VDK value was None (not set) => vdk_log_level = None => the non-root loggers depending on this value were None. Therefore, I added the same vdk-core setting of initial log level by the env vars and everything worked out well. However, if it would work without this, it would be better, I agree.

Reply:
This came from core and we don't need it here. It has to do with logging before vdk is initialized and all the plugins loaded. I removed it and move the calls to this method after the root logger is configured. Should be fine.